### PR TITLE
fix MacOS CI

### PR DESCRIPTION
**Pull request by bashbaug**
The GitHub actions macos-latest workflow was updated to macos-12 recently, which only has gcc-11.  Switch back to macos-11 for now at least, to unblock MacOS CI.

Also updates the checkout action to v3 to eliminate CI warnings.

edit: macos-12 only has gcc-11, not gcc-12.

### CODE REVIEW

**Comment by Kerilk** [comment]
Thanks for fixing it! I supposed it was the issue, but GitHub action documentation is outdated and still lists macos-11 as latest. Apparently it is also not updated for everyone, as I am still getting macos-11 for my fork. I'll approve once I see a working test on MacOS.

**Comment by Kerilk** [comment]
Got to love github actions: "Error: Unable to resolve action `actions/checkout@3`, unable to find version `3`" but only for MacOS
Maybe rollback to `actions/checkout@2` for MacOS only?

**Comment by bashbaug** [comment]
Yeah, I'm pretty sure this is it.  I'm not sure exactly when the switch happened and it still looks like the docs are catching up, but there is some information and the announcement here:

https://github.com/actions/runner-images/issues/6384

I had a typo in my change to update the checkout actions (now fixed), but you can see the previous commit was successful.  I'm pretty confident this latest update should be clean.

**Comment by Kerilk** [comment]
Looks good, I'll let it run approve and merge, this will unblock the rest.

**Review by Kerilk — APPROVED** [review]

**Pull request merged**
